src/signer.js

```diff
@@ -288,6 +300,7 @@ function getCanonicalQueryString(url) {
 function getCanonicalHeaders(headers) {
   var aggregatedHeaders = new Array();
   for (var i=0; i<headers.length; i++) {
+	headers[i].value = headers[i].value.trim().replace(/  +/g, ' ');
```

Comment:
This addresses AWS's rule that consecutive spaces in header values should be collapsed to one space.